src/Compilers/CSharp/Portable/Symbols/TypeSymbolWithAnnotations.cs

```diff
@@ -32,6 +32,43 @@ public static bool IsAnyNotNullable(this NullableAnnotation annotation)
             return annotation == NullableAnnotation.NotAnnotated || annotation == NullableAnnotation.NotNullable;
         }
 
+        public static bool IsSpeakable(this NullableAnnotation annotation)
+        {
+            return annotation == NullableAnnotation.Unknown ||
+                annotation == NullableAnnotation.NotAnnotated ||
+                annotation == NullableAnnotation.Annotated;
+        }
+
+        /// <summary>
+        /// This method projects nullable annotations onto a smaller set that can be expressed in source.
+        /// </summary>
+        public static NullableAnnotation AsSpeakable(this NullableAnnotation annotation, TypeSymbol type)
+        {
+            switch (annotation)
+            {
+                case NullableAnnotation.Unknown:
+                case NullableAnnotation.NotAnnotated:
+                case NullableAnnotation.Annotated:
+                    return annotation;
+
+                case NullableAnnotation.Nullable:
+                    if (type.IsTypeParameterDisallowingAnnotation())
```

Comment:
>type [](start = 24, length = 4)

Is it safe to dereference the ```type```?

Reply:
That's a good question. I spent some time reviewing and testing this to confirm. In short, yes, I think it is safe to dereference.

There are two callers of `AsSpeakable`: `TSWA.AsSpeakable` (which is safe) and a delegate passed into `MethodTypeInferrer.Infer(...)`.
The delegate in the latter case is used in three places. Two are clearly guarded. 
The last one (`MethodTypeInference.OutputTypeInference`) is not so obvious. It has one caller, and that one is guarded by a check (`HasUnfixedParamInOutputType`) which can only pass for an unbound lambda or a method group. Both of those should never get `NullableAnnotation.Nullable`.

---
In reply to: [243095731](https://github.com/dotnet/roslyn/pull/31813#discussion_r243095731) [](ancestors = 243095731)

Reply:
> I spent some time reviewing and testing this to confirm. In short, yes, I think it is safe to dereference.

Consider adding a null check to make code robust to future changes. Or add an assert at the top of the function that ```type``` is never null. 

---
In reply to: [243135384](https://github.com/dotnet/roslyn/pull/31813#discussion_r243135384) [](ancestors = 243135384,243095731)